massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -274,7 +431,7 @@ impl SpeculativeLedger {
         &mut self,
         addr: &Address,
         key: Vec<u8>,
-        data: Vec<u8>,
+        value: Vec<u8>,
     ) -> Result<(), ExecutionError> {
```

Comment:
in this set_data_entry function as well we should charge the caller_address for storage costs

Reply:
(set_datastore_value_for ABI)

Reply:
I don't get it. It's the address that pay for the storage costs. If a person creates a new address to store things in it he send it coins so that the address herself can pay for storage costs.

Reply:
if an addr A is initializing an address B that it just created (A has temporary write access to B), A should be paying / getting storage cost deltas because the call is being done by A

Reply:
I didn't think it will work that way on my mind it was the address that must have the funds and so if someone is creating an address he has to transfer it coins before making storage operations